### PR TITLE
docs(path-resolver): document intentional shared-utility role + edge-case tests

### LINKED ISSUE #90: Add architectural documentation and edge-case tests for PathResolver
## Problem / Motivation

A graphify knowledge graph analysis of the codebase flagged `PathResolver` as a potential god class — it has 29 edges, betweenness centrality 0.127, and is called by 11 introspectors (action_text, active_storage, api, auth, config, controller, model, non_ar_models, stimulus, turbo, view).

## Investigation Findings

After tracing the code, `PathResolver` is **NOT a god class**. It is a well-factored shared utility:

- 229 lines, single clear responsibility (resolve Rails logical paths → filesystem paths safely)
- 3 cohesive private helper classes (`SafeRelativePath`, `SafeJoin`, `ConfiguredPathsError`) — all about path safety
- Good test coverage (8 specs: happy path, traversal rejection, absolute path rejection, error fallback)
- High betweenness is **expected and healthy** for a foundational utility — splitting it would be over-engineering

The graph's flag is a false positive: high betweenness centrality for a shared utility is the correct design, not a smell.

## Expected Outcome / Goal

1. Add architectural documentation explaining `PathResolver`'s intentional role as a shared utility (prevents future "god class" concerns from graph analyses)
2. Add edge-case tests for the private helper classes (`SafeJoin` escape prevention, `SafeRelativePath` Windows path handling) to improve coverage and document their behavior

## Acceptance Criteria

- [ ] `PathResolver` has a class-level docblock explaining its intentional shared-utility role
- [ ] `SafeJoin` behavior is tested directly (escape prevention, valid joins)
- [ ] `SafeRelativePath` Windows path handling is tested
- [ ] All existing tests continue to pass (`bundle exec rspec`)
- [ ] `bundle exec rubocop` passes

Generated with [Devin](https://devin.ai)

### PR DESCRIPTION
**Pull request by igmarin**
## Summary

- **Investigation**: A graphify knowledge graph analysis flagged `PathResolver` as a potential god class (29 edges, betweenness centrality 0.127, 11 introspector callers). After tracing the code, it is **NOT a god class** — it is a well-factored shared utility with a single clear responsibility (resolve Rails logical paths → filesystem paths safely). High betweenness is expected and healthy for a foundational utility.
- **Documentation**: Added a class-level docblock explaining the intentional shared-utility role and why splitting would be over-engineering. This prevents future false positives from graph analyses.
- **Tests**: Added 6 edge-case tests for the private helper classes (`SafeRelativePath`, `SafeJoin`) covering backslash normalization, Windows path rejection, empty path rejection, valid joins, and traversal escape prevention.

Closes #90

## Test plan

- [x] `bundle exec rspec spec/lib/rails_ai_bridge/path_resolver_spec.rb` — 14 examples, 0 failures (8 original + 6 new)
- [x] `bundle exec rubocop lib/rails_ai_bridge/path_resolver.rb spec/lib/rails_ai_bridge/path_resolver_spec.rb` — no offenses
- [ ] Full suite: `bundle exec rspec` (CI will verify)

Generated with [Devin](https://devin.ai)

## Summary by CodeRabbit

* **Documentation**
  * Added documentation clarifying path resolution behavior and safety boundaries.

* **Bug Fixes**
  * Improved validation of relative paths, including Windows-style absolute paths and empty inputs.
  * Prevented path traversal when joining files with a base directory.

* **Tests**
  * Added coverage for path normalization, valid relative paths, absolute path rejection, and directory escape prevention.

### CODE REVIEW

File: spec/lib/rails_ai_bridge/path_resolver_spec.rb [hunk 1]
```diff
@@ -90,4 +90,46 @@
       expect(Rails.logger).to have_received(:error).with(%r{failed to read path "app/models"})
     end
   end
+
+  # Edge-case tests for the private helper classes. These are accessed via
+  # Ruby's constant lookup since they are private_constant — the specs document
+  # their behavior and guard against regressions in the path-safety surface.
+  describe 'SafeRelativePath (private helper)' do
+    let(:safe_relative_path) { described_class.const_get(:SafeRelativePath) }
```
[the remark this answers is not in the document]
Comment (igmarin) [reply spec/lib/rails_ai_bridge/path_resolver_spec.rb:98]: **Pushing back — `const_get` works on private constants.**

This finding is factually incorrect. In Ruby, `private_constant` only prevents **lexical** access (e.g., `RailsAiBridge::PathResolver::SafeRelativePath` raises `NameError`). It does **not** block reflective access via `const_get`.

Proof — verified locally and on CI:

```ruby
class Foo
  class Bar; end
  private_constant :Bar
end

Foo.const_get(:Bar) # => Foo::Bar (works, no NameError)
Foo::Bar            # => NameError: private constant Foo::Bar referenced
```

The CI run confirms this: **2163 examples, 0 failures** — all 6 new edge-case tests pass using `const_get`. If `const_get` raised `NameError`, the tests would fail.

**Why `const_get` is preferable to `class_eval`:**
1. It's the standard Ruby reflection API for accessing constants — more explicit about intent
2. `class_eval('SafeRelativePath')` uses string eval, which is a weaker pattern (semgrep/brakeman flag string eval as a code smell even in tests)
3. `const_get` is what Ruby's own test suite (MRI `test/ruby/test_module.rb`) uses to test `private_constant` behavior

Keeping `const_get` as-is. No change needed.

**Comment by igmarin** [comment]
## CI Status Update

### Lint failure — FIXED ✅
The `Style/ArrayIntersect` offense was **pre-existing on `main`** (in `context_summary.rb:462`, a file not touched by this PR). It was introduced by a newer rubocop version after the last successful CI run on June 30. Fixed with `rubocop -A` autocorrect in commit 7e9dc28.

### Ruby 3.3 / Rails 8.0 failure — PRE-EXISTING ⚠️
This failure is **not caused by this PR**. The same failure occurs on:
- `main` branch (last successful CI: June 30)
- `dependabot/bundler/simplecov-tw-1.0` branch (July 18): same `2163 examples, 0 failures` but `exit code 1`

All tests pass (`0 failures`) — the non-zero exit code comes from an environmental issue with Rails 8.0 + Ruby 3.3, not from any test failure. This should be tracked as a separate issue.

### CodeRabbit review — pushed back 🔄
The `CHANGES_REQUESTED` comment claims `const_get` raises `NameError` for `private_constant`. This is incorrect — `const_get` works on private constants (only lexical access is blocked). See my reply on the inline comment for evidence. All 6 new tests pass on CI confirming this.
